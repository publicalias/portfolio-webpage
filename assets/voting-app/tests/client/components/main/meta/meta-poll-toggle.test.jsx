"use strict";

//local imports

const MetaPollToggle = require("../../../../../scripts/client/components/main/meta/meta-poll-toggle");

const { newPoll } = require("../../../../../schemas");
const { testWrapper } = require("../../../test-helpers");

//global imports

const { capitalize } = require("all/utilities");
const { newUser } = require("redux/schemas");
const { initTestEvent, initTestSnapshot, withDataList } = require("redux/tests/client-tests");
const { reactTests } = require("redux/tests/react-tests");

//utilities

const { testShallow } = testWrapper(MetaPollToggle);

const testRole = (render, role, prop, type = capitalize(role)) => {

  const testSnapshot = initTestSnapshot(render);

  const testClick = initTestEvent(render, "click");

  it("should match snapshot (default)", () => testSnapshot());

  it("should match snapshot (toggled", () => testSnapshot({ user: newUser({ id: "id-a" }) }, {
    poll: {
      users: {
        [prop]: ["id-a"]
      }
    }
  }));

  it("should match snapshot (util)", () => testSnapshot(null, { util: "u-meta" }));

  it("should call handleClick on click", () => testClick(
    `.qa-toggle-${role}`,
    [null, { poll: { id: "id-a" } }],
    [`pollToggle${type}`, ["id-a"]]
  ));

};

//setup

beforeAll(reactTests.setup);
beforeEach(reactTests.inject(MetaPollToggle));

//meta poll toggle

describe("MetaPollToggle (flag)", () => {

  const testFlag = withDataList(testShallow, [null, {
    poll: newPoll(),
    role: "flag"
  }]);

  testRole(testFlag, "flag", "flagged");

});

describe("MetaPollToggle (hide)", () => {

  const testHide = withDataList(testShallow, [null, {
    poll: newPoll(),
    role: "hide"
  }]);

  testRole(testHide, "hide", "hidden");

});
