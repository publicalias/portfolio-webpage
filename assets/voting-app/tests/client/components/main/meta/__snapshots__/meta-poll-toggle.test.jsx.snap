// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetaPollToggle (flag) should match snapshot (default) 1`] = `
<button
  className="c-icon-button qa-toggle-flag "
  onClick={[Function]}
>
  <i
    className="far fa-flag"
  />
</button>
`;

exports[`MetaPollToggle (flag) should match snapshot (toggled 1`] = `
<button
  className="c-icon-button qa-toggle-flag "
  onClick={[Function]}
>
  <i
    className="fas fa-flag"
  />
</button>
`;

exports[`MetaPollToggle (flag) should match snapshot (util) 1`] = `
<button
  className="c-icon-button qa-toggle-flag u-meta"
  onClick={[Function]}
>
  <i
    className="far fa-flag"
  />
</button>
`;

exports[`MetaPollToggle (hide) should match snapshot (default) 1`] = `
<button
  className="c-icon-button qa-toggle-hide "
  onClick={[Function]}
>
  <i
    className="far fa-eye"
  />
</button>
`;

exports[`MetaPollToggle (hide) should match snapshot (toggled 1`] = `
<button
  className="c-icon-button qa-toggle-hide "
  onClick={[Function]}
>
  <i
    className="far fa-eye-slash"
  />
</button>
`;

exports[`MetaPollToggle (hide) should match snapshot (util) 1`] = `
<button
  className="c-icon-button qa-toggle-hide u-meta"
  onClick={[Function]}
>
  <i
    className="far fa-eye"
  />
</button>
`;
